src/fluree/db/ledger/transact.clj

```diff
@@ -621,7 +621,9 @@
         ;; go through each statement and check
         (loop [[^Flake flake & r] flakes]
           (when (> (.-s flake) const/$maxSystemPredicates)
-            (when-not (<? (perm-validate/allow-flake? candidate-db flake tx-permissions))
+            (when-not (if (.-op flake)
```

Comment:
The permission check here could go into its own function, but it's probably not worth it since this file will be refactored soon.

Reply:
Hopefully soon.